prereise/gather/griddata/transmission/geometry.py

```diff
@@ -0,0 +1,273 @@
+from dataclasses import dataclass
+from math import exp, log, pi, sqrt
+from statistics import geometric_mean
+
+from prereise.gather.griddata.transmission.const import (
+    epsilon_0,
+    mu_0,
+    relative_permeability,
+    resistivity,
+)
+
+
+@dataclass
+class Conductor:
+    """Represent a single conductor (which may be a stranded compsite).
+
+    :param float radius: outer radius of conductor.
+    :param str material: material of conductor. Used to calculate ``resistance_per_km``
+        and ``gmr`` if these aren't passed to the constructor, unnecessary otherwise.
+    :param float resistance_per_km: outer radius of conductor. Will be estimated from
+        other parameters if it isn't passed.
+    :param float gmr: geometric mean radius of conductor. Will be estimated from
+        other parameters if it isn't passed.
+    :param float area: cross-sectional area of conductor. Will be estimated from
+        other parameters if it isn't passed.
+    """
+
+    radius: float
+    material: str = None
+    resistance_per_km: float = None
+    gmr: float = None
+    area: float = None
+
+    def __post_init__(self):
+        # Validate inputs
+        if self.gmr is None and (self.material is None or self.radius is None):
```

Comment:
According to the definition, `self.radius` is a positional argument, which can't be `None` or must be provided?

Reply:
Good catch, fixed. That was from an older iteration of the code where I didn't realize that I always need radius in order to be able to calculate capacitance.